sentry-ruby/spec/sentry/client/event_sending_spec.rb

```diff
@@ -34,6 +34,7 @@
 
         it "doesn't send the event when it's not sampled" do
           allow(Random).to receive(:rand).and_return(0.51)
+          expect(subject.transport).to receive(:record_lost_event).with(:sample_rate, 'event')
```

Comment:
can we test this (and below scenarios) end-to-end instead of using mocking? you can even write a custom helper like `have_recorded_lost_event`. so it'll be like:

```rb
# this will check the value in @discarded_events
expect(transport).to have_recorded_lost_event(:sample_rate, 'event') 
```